test/cppsuite/how_to_use_cppsuite.md

```diff
@@ -0,0 +1,175 @@
+
+#### Defining workloads
+In our example test `multi_size_inserts` we will first update the functions that define `insert` and `populate`.
+The default behavior of populate will create the table in the database, so we can delete the stubbed implementation of `populate` and make sure that it is defined in the [configuration file](#the-configuration-file) later.
+When cppsuite executes the test it will create threads that run the `insert_operation` function.  
+
+Initially, the insert function is stubbed as follows:
+```cpp
+void
+insert_operation(thread_worker *) override final
+{
+    logger::log_msg(LOG_WARN, "insert_operation: nothing done");
+}
+```
+The user can either define their own insertion logic by filling out the function body, or remove the `insert_operation` stub to fall back on the framework’s default implementation.
+For our `multi_size_inserts` example, we add code to to write transactions with either 10MB or 1KB updates. We will configure the `cache_size` to be 10MB later.
+
+```cpp
+void
+insert_operation(thread_worker *tw) override final
+{
+    // Retrieve the collection created by the populate operation.
+    collection &coll = tw->db.get_collection(0);
+    scoped_cursor cursor = tw->session.open_scoped_cursor(coll.name);
+
+    int value_size_bytes = 1000;
+
+    while (tw->running()) {
+
+        // Swap value size between 1KB and 10MB.
+        value_size_bytes = value_size_bytes == 1000 ? 10000000 : 1000;
+
+        tw->txn.begin();
+
+        bool success = true;
+        // Each transaction should have target_op_count insert operations. This value is set in the configuration file
+        for(int i = 0; i < tw->txn.get_target_op_count(); ++i) {
+            tw->sleep();
+    
+            // Populate the key and value with random strings. We only use the size of the value in this test
+            const std::string key = random_generator::instance().generate_pseudo_random_string(tw->key_size);
+            const std::string value = random_generator::instance().generate_pseudo_random_string(value_size_bytes);
+
+            if(!tw->insert(cursor, coll.id, key, value)) {
+                success = false;
+                break;
+            }
+        }
+
+        if(!success){
+            tw->txn.rollback();
+        } else {
+            testutil_assert(tw->txn.commit());
+        }
+    }
+}
+```
+
+
+#### Saving validation data
+The `set_tracking_cursor` function defines what data is saved by the operation tracker. 
+In our example test we will save the timestamp, transaction ID, key and value size.
+
+```cpp
+void
+set_tracking_cursor(WT_SESSION *session, const tracking_operation &operation, const uint64_t &collection_id,
+    const std::string &key, const std::string &value, wt_timestamp_t ts,
+    scoped_cursor &op_track_cursor) override final
+{
+    uint64_t txn_id = ((WT_SESSION_IMPL *)session)->txn->id;
```

Comment:
In C++, it would be better to use a static_cast<> rather than a C-style cast. See [the Mongo Server Code Style](https://github.com/mongodb/mongo/wiki/Server-Code-Style#casting)

Reply:
Updated in a1ee228bf1c057be8095493025bd6ae21c61d804. I needed to use a `reinterpret_cast` as session and session_impl are just C structs and `static_cast` throws a `not related by inheritance` error.

Reply:
Fair point. The fact that that you had to use `reinterpret_cast<>` makes the sort of cast that's required much more explicit than when using C-style casts. Thanks for making the change.